### PR TITLE
Update node-resque types to match v5.5.7.

### PR DESCRIPTION
**Pull request by petetnt**
Signed-off-by: petetnt <pete.a.nykanen@gmail.com>

This PR updates the `node-resque` types to match the current version 5.5.7. that seems to have quite a lot of new methods but not much breaking changes.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/taskrabbit/node-resque
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/node-resque/index.d.ts [hunk 1]
```diff
@@ -36,14 +36,46 @@ export interface QueueOptions {
     connection?: ConnectionOptions;
 }
 
+type WorkerStatus = {
```
Comment (gordey4doronin) [inline types/node-resque/index.d.ts]: I think using interfaces over types is still preferred in that repository?
BTW That's one of the reasons why the build fails.

File: types/node-resque/index.d.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
-// Type definitions for node-resque 5.4
+// Type definitions for node-resque 5.5.7
```
Comment (gordey4doronin) [inline types/node-resque/index.d.ts]: The build says patch version is not allowed. Honestly I'm not sure why, would be interested to know. But if you want build to pass you probably need to change it. 😬

**Comment by petetnt** [comment]
Thanks for the early review @gordey4doronin ! The bot was a bit hasty, published this one as a draft yesterday as I didn't have the time to finish it yesterday but the bot still went ahead and did all its tasks 😅. I'll do some additions + tests today and ping you after that.

File: types/node-resque/index.d.ts (hunk 2 above)
Comment (petetnt) [reply types/node-resque/index.d.ts]: Yeah makes sense, patches shouldn't contain API changes (not that semver is set in stone). I'll change those!

**Comment by gfoltz** [comment]
The `ConnectionOptions` interface also needs an optional `redis` attribute.

See https://github.com/taskrabbit/node-resque/blob/master/lib/connection.js#L41
And the readme at https://github.com/taskrabbit/node-resque/blame/master/README.md#L164-L170

**Comment by petetnt** [comment]
Fixed some of the types, added tests and cleaned up linting errors.

I added the `redis?` parameter as `any`, it should be `redis?: IORedis` but referencing it with `/// references types="ioredis"` has some external errors that should be fixed...

**Comment by petetnt** [comment]
The BenchmarkPR test failed as `typescript@3.7` is not out yet, right? ¯\_(ツ)_/¯

**Review by gordey4doronin — APPROVED** [review]

File: types/node-resque/node-resque-tests.ts [hunk 3]
```diff
@@ -64,9 +64,41 @@ worker.connect().then(() => worker.start());
 // start scheduler
 scheduler.connect().then(() => scheduler.start());
 
-queue.connect().then(() => {
+queue.connect().then(async () => {
```
Comment (gordey4doronin) [inline types/node-resque/node-resque-tests.ts]: Can we also add the following snippet after line 24, similar to #40027 to make it fully consistent to @gfoltz changes.

```ts
 const connectionWithCustomRedis: ConnectionOptions = { redis: "fake_redis_client" };
```

**Comment by gordey4doronin** [comment]
Here is an issue for failing build https://github.com/DefinitelyTyped/DefinitelyTyped/issues/40054 🤔

File: types/node-resque/node-resque-tests.ts (hunk 3 above)
Comment (petetnt) [reply types/node-resque/node-resque-tests.ts]: Updated in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/39973/commits/45a3df7a5e9f1801a0722d4857f842d4dced6b99

Comment (gordey4doronin) [reply types/node-resque/node-resque-tests.ts]: 👍 Hopefully failing CI would be figured out soon and you can merge then. 🙏

**Pull request merged**